packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -1022,3 +1077,19 @@ const findOldestCallEnded = (calls: { [key: string]: { endTime?: Date } }): stri
   }
   return oldestCallId;
 };
+
+/* @conditional-compile-remove(reaction) */
+function clearParticipantReactionState(callContext: CallContext, callId: string, participantKey: string) {
+  callContext.setReceivedReactionFromParticipant(callId, participantKey, null);
+}
+
+function isReactionBeingPlayedNow(receivedAt: Date | undefined) {
+  if (receivedAt == undefined) {
+    return false;
+  }
+  const currentTimeStamp = new Date();
+  const unixBasedCurrentTimeStamp = Math.floor(currentTimeStamp.getTime() / 1000);
+  const unixBasedReceivedTimeStamp = Math.floor(receivedAt?.getTime() / 1000);
+
+  return receivedAt != undefined && unixBasedCurrentTimeStamp - unixBasedReceivedTimeStamp < 3000;
```

Comment:
nit: already checked `receivedAt != undefined`

```suggestion
  return unixBasedCurrentTimeStamp - unixBasedReceivedTimeStamp < 3000;
```